daemon/api_model_test.go

```diff
@@ -251,3 +291,67 @@ func (s *apiSuite) TestGetModelHasSerialAssertion(c *check.C) {
 	c.Assert(devKey, check.FitsTypeOf, "")
 	c.Assert(devKey.(string), check.Equals, string(encDevKey))
 }
+
+func (s *apiSuite) TestGetModelJSONHasSerialAssertion(c *check.C) {
+	// make a model assertion
+	theModel := s.brands.Model("my-brand", "my-old-model", modelDefaults)
+
+	deviceKey, _ := assertstest.GenerateKey(752)
+
+	encDevKey, err := asserts.EncodePublicKey(deviceKey.PublicKey())
+	c.Assert(err, check.IsNil)
+
+	// model assertion setup
+	d := s.daemonWithOverlordMock(c)
+	hookMgr, err := hookstate.Manager(d.overlord.State(), d.overlord.TaskRunner())
+	c.Assert(err, check.IsNil)
+	deviceMgr, err := devicestate.Manager(d.overlord.State(), hookMgr, d.overlord.TaskRunner(), nil)
+	c.Assert(err, check.IsNil)
+	d.overlord.AddManager(deviceMgr)
+	st := d.overlord.State()
+	st.Lock()
```

Comment:
best to do a defer st.Unlock()  as well when locking/unlocking over many checks, otherwise it's easy to get test failures that will panic on lock sanity instead of giving the actual problem